src/GafferImage/Display.cpp

```diff
@@ -228,23 +225,59 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
 			return true;
 		}
 
-		ConstFloatVectorDataPtr channelData( const Imath::V2i &tileOrigin, const std::string &channelName )
+		ConstFloatVectorDataPtr channelData( const Imath::V2i &tileOrigin, const std::string &channelName, int dataCount )
 		{
 			vector<string>::const_iterator cIt = find( channelNames().begin(), channelNames().end(), channelName );
 			if( cIt == channelNames().end() )
 			{
 				return ImagePlug::blackTile();
 			}
 
-			ConstFloatVectorDataPtr tile = getTile( tileOrigin, cIt - channelNames().begin() );
-			if( tile )
+			Tile *tile = getTile( tileOrigin, cIt - channelNames().begin() );
+			if( !tile )
+			{
+				return ImagePlug::blackTile();
+			}
+
+			int previousDataCount;
+
+			{
+				tbb::spin_rw_mutex::scoped_lock tileLock( tile->mutex, false /* read */ );
+
+				previousDataCount = tile->cachedForDataCount;
+
+				if( tile->cachedForDataCount == dataCount || !tile->dirty)
+				{
+					// In order to ensure hashes and computes are consistent, once we have
+					// bound a version of the tile for this dataCount, we don't change it,
+					// even if it has been dirtied in the meantime.
+					return tile->cachedTile;
+				}
+			}
+
+			// This is labelled as const just so we can swap it with the cachedTile
+			ConstFloatVectorDataPtr newCache = new FloatVectorData();
+			const_cast<FloatVectorData*>( newCache.get() )->writable() = tile->backBuffer;
+
+			tbb::spin_rw_mutex::scoped_lock tileLock( tile->mutex, true /* write */ );
```

Comment:
With the simplification above, we could just reuse the original tileLock with `tileLock.acquire()`.

Reply:
Addressed.